### DISCUSSION TITLE
The static directory has been deprecated in favor of the public directory. https://err.sh/vercel/next.js/static-dir-deprecated

### DISCUSSION THREAD

**Discussion by Er-Ramanand-Dubey** [vercel/next.js#14018] [Help] [answered]
** I am trying to update Nextjs version in my project ..but I am getting this error can anyone help me**

(node:12420) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'unshift' of undefined
    at cfg.entry (\next.config.js:51:31)
    at async DynamicEntryPlugin.webpackConfig.entry (\node_modules\next\dist\build\webpack-config.js:145:156)
(node:12420) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:12420) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> Using external babel configuration
> Location: "\.babelrc"

**Comment by jamesmosier** [comment, marked as answer]
Could you share some code? 😄 it looks like line 51 in your `next.config.js` is causing the issues.

**Reply by Er-Ramanand-Dubey** [reply]
//this code belongs to 46 to 58 line in next.config.js file..please let me know
	webpack: function (cfg) {
			    const originalEntry = cfg.entry
			    cfg.entry = async () => {
						  const entries = await originalEntry()
				    			entries['main.js'].unshift('./client/polyfills.js')
				          return entries
				        }

			    return cfg
				}

**Reply by jamesmosier** [reply]
Take a look at this polyfills example: https://github.com/vercel/next.js/blob/canary/examples/with-polyfills/next.config.js

You'll want to do a conditional around your `unshift`:

```js
      if (
        entries['main.js'] &&
        !entries['main.js'].includes('./client/polyfills.js')
      ) {
        entries['main.js'].unshift('./client/polyfills.js')
      }
```

**Reply by Er-Ramanand-Dubey** [reply]
thank u so much this is works...but Now i am got another error
error - ./client/polyfills.js
TypeError: \client\polyfills.js: this.setDynamic is not a function

**Reply by jamesmosier** [reply]
If you could provide a full code sample that'd be helpful. It is hard to say without knowing where `setDynamic` is or what it does.

**Reply by Er-Ramanand-Dubey** [reply]
which file ?

**Reply by jamesmosier** [reply]
A good start would be the file that the error is occurring in: `\client\polyfills.js`

**Reply by Er-Ramanand-Dubey** [reply]
okay ..this is whole code of file \client\polyfills.js


import includes from 'core-js/library/fn/string/virtual/includes'

import repeat from 'core-js/library/fn/string/virtual/repeat'

import assign from 'core-js/library/fn/object/assign'



import from from 'core-js/library/fn/array/from'



console.log('Load your polyfills')



String.prototype.includes = includes

String.prototype.repeat = repeat

Object.assign = assign

Array.from = from

**Reply by jamesmosier** [reply]
I'm not sure from looking at that.

You can probably get rid of this custom polyfill altogether. Next includes these polyfills by default.

Here is the complete list: https://github.com/vercel/next.js/blob/canary/packages/next-polyfill-nomodule/src/index.js
